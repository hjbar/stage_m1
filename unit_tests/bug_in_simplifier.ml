let () =
  let module Constraint = Constraint.Make(MSeq) in
  let module Solver = Solver.Make(MSeq) in
  let c : _ Constraint.t =
    let open Constraint in
    let a = Var.fresh "a" in 
    let b = Var.fresh "b" in 
    Exist (a, None, 
    Exist (b, None, 
      Conj ( 
        Eq (a, b),
        Do MSeq.fail )))
  in
  let (logs, _env, _result) =
    Solver.eval ~log:true Unif.Env.empty c in
  PPrint.(separate hardline logs)
  |> Utils.string_of_doc
  |> print_endline
  |> print_newline

