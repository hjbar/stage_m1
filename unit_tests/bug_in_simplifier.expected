∃?a ?b. do? ∧ ?a = ?b
∃?a ?b. do? ∧ ?a = ?b
∃?b ?a. do? ∧ ?a = ?b
∃?a. do?

